synapse/lib/base.py

```diff
@@ -304,8 +320,6 @@ def _iAmLoop(self):
             except Exception:
                 # The taskDone callback will emit the exception.  No need to repeat
                 pass
-        await asyncio.sleep(0, loop=self.loop)
```

Comment:
this sleep(0) calls gives all the tasks which were just cancelled the chance to clean up.  Without it, it is possible (I believe) on interpreter shutdown that we can end up emitting warnings about unfinished tasks kicking around.

Reply:
Good catch.

Reply:
sounds right.  adding back.